### PR TITLE
Remove CuPy document for v2

### PR DESCRIPTION
**Pull request by beam2d**
This PR removes the documentations related to CuPy, which will be a separate project from v2.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -91,70 +91,35 @@ That shows all logs of installation. It may helps you::
...
 .. _install_cuda:
 
-Install Chainer with CUDA
+Enable CUDA/cuDNN support
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-You need to install CUDA Toolkit before installing Chainer.
-If you have CUDA in a default directory or set ``CUDA_PATH`` correctly, Chainer installer finds CUDA automatically::
+In order to enable CUDA support, you first have to install `CuPy <https://docs.cupy-ndarray.org/>`_.
+It requires CUDA Toolkit and ``CUDA_PATH`` environment variable.
```
Comment (unnonouno) [inline docs/source/install.rst]: This environment variable is only for installation. Do we need to write condition about installation here?

File: docs/source/install.rst [hunk 2]
```diff
@@ -91,70 +91,35 @@ That shows all logs of installation. It may helps you::
...
+If CuPy is correctly installed, Chainer will automatically enable CUDA support::
```
Comment (unnonouno) [inline docs/source/install.rst]: Installation method of CuPy will be written in CuPy document. Is a detailed description required?

File: docs/source/install.rst (hunk 1 above)
Comment (beam2d) [inline docs/source/install.rst]: If CuPy's installation guide is well written, we may only need to let users read them.

File: docs/source/install.rst (hunk 2 above)
Comment (beam2d) [inline docs/source/install.rst]: OK, I think it's not needed. I will remove them.

**Comment by beam2d** [comment]
I will fix the install guide related to cupy after CuPy documentation is made ready.

**Comment by unnonouno** [comment]
OK, let's fix cupy related issues in another pr.

**Pull request merged**